birdhouse/config/jupyterhub/jupyterhub_config.py.template

```diff
@@ -1,21 +1,60 @@
 import os
+from os.path import join
 import logging
+import subprocess
+from shutil import copy
 
 c.JupyterHub.bind_url = 'http://:8000/jupyter'
 c.JupyterHub.hub_ip = 'jupyterhub'
 
+c.JupyterHub.authenticator_class = 'jupyterhub_magpie_authenticator.MagpieAuthenticator'
+c.MagpieAuthenticator.magpie_url = "http://magpie:2001"
+
 c.JupyterHub.cookie_secret_file = '/persist/jupyterhub_cookie_secret'
 c.JupyterHub.db_url = '/persist/jupyterhub.sqlite'
 
-c.JupyterHub.template_paths = ['/template_paths']
+c.JupyterHub.template_paths = ['/custom_templates']
 
 c.JupyterHub.spawner_class = 'dockerspawner.DockerSpawner'
+
 c.DockerSpawner.image = os.environ['DOCKER_NOTEBOOK_IMAGE']
 c.DockerSpawner.use_internal_ip = True
 c.DockerSpawner.network_name = os.environ['DOCKER_NETWORK_NAME']
+
 notebook_dir = '/notebook_dir'
+jupyterhub_data_dir = os.environ['JUPYTERHUB_USER_DATA_DIR']
+host_user_data_dir = join(jupyterhub_data_dir, "{username}")
+container_workspace_dir = join(notebook_dir, "writable-workspace")
+
 c.DockerSpawner.notebook_dir = notebook_dir
-c.DockerSpawner.volumes = { os.environ['DOCKER_JUPYTERHUB_USER_PERSISTENCE_VOLUME']: notebook_dir }
+
+c.DockerSpawner.volumes = {host_user_data_dir: container_workspace_dir}
+
+host_tutorial_notebooks_dir = join(jupyterhub_data_dir, "tutorial-notebooks")
+if os.path.exists(host_tutorial_notebooks_dir):
+    c.DockerSpawner.volumes[host_tutorial_notebooks_dir] = {
+        "bind": join(notebook_dir, "tutorial-notebooks"), 
```

Comment:
Will have to test tomorrow, bind mount might refresh only if container is restarted.  Typical user do not know how to restart their jupyter server.

Reply:
Good catch, see https://github.com/bird-house/birdhouse-deploy/pull/16/commits/6ddaddc74d384299e45b0dc8d50a63e59b3cc0d5